src/wallet/wallet.cpp

```diff
@@ -100,7 +100,7 @@ CPubKey CWallet::GenerateNewKey()
 
     // use HD key derivation if HD was enabled during wallet creation
     if (IsHDEnabled()) {
-        DeriveNewChildKey(metadata, secret);
+        DeriveNewChildKey(metadata, secret, (CanSupportFeature(FEATURE_HD_SPLIT) ? internal : false));
```

Comment:
Is there a reason to do this check here, rather than inside `DeriveNewChildKey`?